### PR TITLE
[WFLY-15542] Connector subsystem Jakarta migration

### PR DESCRIPTION
**Pull request by tadamski**
https://issues.redhat.com/browse/WFLY-15542

Depends on:
https://github.com/wildfly/wildfly/pull/14975

### CODE REVIEW

File: ee-9/pom.xml [hunk 1]
```diff
@@ -416,6 +417,18 @@
...
+                <groupId>jakarta.security.auth.message</groupId>
+                <artifactId>jakarta.security.auth.message-api</artifactId>
+                <version>${version.jakarta.security.auth.message-api}</version>
+                <exclusions>
+                    <exclusion>
+                        <groupId>*</groupId>
+                        <artifactId>*</artifactId>
+                    </exclusion>
+                </exclusions>
+            </dependency>
```
Comment (yersan) [inline ee-9/pom.xml:430]: This dependency should be `jakarta.authentication:jakarta.authentication-api:2.0.0` instead, which is already available in the pom, so I would not expect to introduce this variant (2.0.0-RC1) here.

**Review by yersan — CHANGES_REQUESTED** [review]

File: ee-9/source-transform/connector/pom.xml [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.wildfly</groupId>
+        <artifactId>wildfly-ee-9-source-transform-parent</artifactId>
+        <!--
+        Maintain separation between the artifact id and the version to help prevent
+        merge conflicts between commits changing the GA and those changing the V.
+        -->
+        <version>26.0.0.Final-SNAPSHOT</version>
```
Comment (darranl) [inline ee-9/source-transform/connector/pom.xml:34]: I think this will need updating to the current version in use on main.

**Comment by darranl** [comment]
@tadamski I think this one still needs some updates before we test again.

**Comment by tadamski** [comment]
@darranl I plan to fix and rebase this PR after transactions migration is merged.

**Comment by bstansberry** [comment]
@tadamski The transactions subsystem one is merged now.

File: ee-9/pom.xml (hunk 1 above)
Comment (tadamski) [reply ee-9/pom.xml]: fixed; I found out that this can be removed

File: ee-9/source-transform/connector/pom.xml (hunk 2 above)
Comment (tadamski) [reply ee-9/source-transform/connector/pom.xml]: fixed

**Comment by tadamski** [comment]
@bstansberry I have rebased and fixed the PR; should work fine now

**Review by yersan — CHANGES_REQUESTED** [review]

File: ee-9/pom.xml [hunk 3]
```diff
@@ -771,6 +771,103 @@
...
+                <groupId>org.jboss.ironjacamar</groupId>
+                <artifactId>ironjacamar-core-impl-jakarta</artifactId>
+                <version>${version.org.jboss.ironjacamar}</version>
+                <!--FIXME removing this results in NPE during compilation but I can't see any particular dependency blocked by the exclusion-->
+<!--                <exclusions>-->
+<!--                    <exclusion>-->
+<!--                        <groupId>*</groupId>-->
+<!--                        <artifactId>*</artifactId>-->
+<!--                    </exclusion>-->
+<!--                </exclusions>-->
```
[the remark this answers is not in the document]
Comment (tadamski) [reply ee-9/pom.xml]: fixed

File: ee-9/pom.xml [hunk 4]
```diff
@@ -771,6 +771,103 @@
...
+                <groupId>org.jboss.ironjacamar</groupId>
+                <artifactId>ironjacamar-validator-jakarta</artifactId>
+                <version>${version.org.jboss.ironjacamar}</version>
+                <exclusions>
+                    <exclusion>
+                        <groupId>*</groupId>
+                        <artifactId>*</artifactId>
+                    </exclusion>
+                </exclusions>
+            </dependency>
```
[the remark this answers is not in the document]
Comment (tadamski) [reply ee-9/pom.xml]: fixed

File: ee-9/pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (tadamski) [reply ee-9/pom.xml]: fixed

File: ee-9/pom.xml [hunk 5]
```diff
@@ -1284,12 +1381,12 @@
...
-                        <groupId>*</groupId>
-                        <artifactId>*</artifactId>
-                    </exclusion>
-                </exclusions>
+<!--                <exclusions>-->
+<!--                    <exclusion>-->
+<!--                        <groupId>*</groupId>-->
+<!--                        <artifactId>*</artifactId>-->
+<!--                    </exclusion>-->
+<!--                </exclusions>-->
```
[the remark this answers is not in the document]
Comment (tadamski) [reply ee-9/pom.xml]: fixed

File: ee-9/source-transform/connector/pom.xml [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+        <artifactId>wildfly-ee-9-source-transform-parent</artifactId>
+        <!--
+        Maintain separation between the artifact id and the version to help prevent
+        merge conflicts between commits changing the GA and those changing the V.
+        -->
+        <version>27.0.0.Beta1-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>wildfly-connector-jakarta</artifactId>
```
[the remark this answers is not in the document]
Comment (tadamski) [reply ee-9/source-transform/connector/pom.xml:38]: fixed

**Comment by tadamski** [comment]
@yersan thanks for all the feedback - I have fixed the problems that you mentioned above

@fjuma @darranl IronJacamar still relies on PicketBox. Because of that, I had to temporary restore PicketBox dependency to make ee-9 connector compile:
https://github.com/tadamski/wildfly/commit/8cd99816b30a2db5d69a28831f9c1b739709c638#diff-31d9e3bb0671159c056658b3acb69396cbcf1a43a00db007c39c99817177c190R262

there is a integration layer that I think should be migrated to Elytron:
https://github.com/ironjacamar/ironjacamar/tree/main/core/impl/src/main/java/org/jboss/jca/core/security/picketbox

I have created https://issues.redhat.com/browse/JBJCA-1441 to deal with this issue.  

The question is whether it should be implemented before this PR is fixed. For now, I have updated the branch so that it can be merged before the fix.

File: ee-9/feature-pack/pom.xml [hunk 7]
```diff
@@ -779,6 +779,38 @@
...
                     <artifactId>istack-commons-tools</artifactId>
                 </exclusion>
+                <exclusion>
+                    <groupId>org.jboss.ironjacamar</groupId>
+                    <artifactId>ironjacamar-common-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.jboss.ironjacamar</groupId>
+                    <artifactId>ironjacamar-common-impl-jakarta</artifactId>
+                </exclusion>
```
Comment (yersan) [inline ee-9/feature-pack/pom.xml:789]: Hi @tadamski , you have to exclude the old non -jakarta versions from this feature pack.

**Review by yersan — CHANGES_REQUESTED** [review]

File: ee-9/feature-pack/pom.xml (hunk 7 above)
Comment (tadamski) [reply ee-9/feature-pack/pom.xml]: hi @yersan, fixed

File: ee-9/pom.xml [hunk 8]
```diff
@@ -1425,6 +1521,18 @@
...
+                <groupId>${ee.maven.groupId}</groupId>
+                <artifactId>wildfly-security-plugins</artifactId>
+                <version>${ee.maven.version}</version>
+                <exclusions>
+                    <exclusion>
+                        <groupId>*</groupId>
+                        <artifactId>*</artifactId>
+                    </exclusion>
+                </exclusions>
+            </dependency>
```
Comment (bstansberry) [inline ee-9/pom.xml:1534]: This dependency no longer exists.

Comment (tadamski) [reply ee-9/pom.xml]: fixed

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @tadamski! :) And @yersan for the review

**Pull request merged**